src/targets/gpu/CMakeLists.txt

```diff
@@ -22,43 +22,135 @@
+    kernels/include/migraphx/kernels/reduce.hpp
+    kernels/include/migraphx/kernels/roialign.hpp
+    kernels/include/migraphx/kernels/scatternd.hpp
+    kernels/include/migraphx/kernels/shape.hpp
+    kernels/include/migraphx/kernels/softmax.hpp
+    kernels/include/migraphx/kernels/tensor_view.hpp
+    kernels/include/migraphx/kernels/type_traits.hpp
+    kernels/include/migraphx/kernels/types.hpp
+    kernels/include/migraphx/kernels/vec.hpp
+    kernels/include/migraphx/kernels/vectorize.hpp)
+
+if(NOT WIN32)
+    include(Embed)
+    add_embed_library(migraphx_kernels ${KERNEL_FILES})
+    target_include_directories(migraphx_kernels
+        PRIVATE
+            ${CMAKE_SOURCE_DIR}/src/include
+            ${CMAKE_BINARY_DIR}/include
+            ${CMAKE_BINARY_DIR}/src/include)
+endif()
 
-file(GLOB DEVICE_GPU_SRCS ${CONFIGURE_DEPENDS} ${CMAKE_CURRENT_SOURCE_DIR}/device/*.cpp)
-add_library(migraphx_device ${DEVICE_GPU_SRCS})
+generate_export_header(migraphx_device
+    EXPORT_FILE_NAME ${CMAKE_BINARY_DIR}/include/migraphx/device/export.h)
 
 add_library(compile_for_gpu INTERFACE)
-target_compile_options(compile_for_gpu INTERFACE -std=c++17 -fno-gpu-rdc -Wno-cuda-compat -Wno-unused-command-line-argument -Xclang -fallow-half-arguments-and-returns)
-target_link_libraries(compile_for_gpu INTERFACE hip::device -fno-gpu-rdc -Wno-invalid-command-line-argument -Wno-unused-command-line-argument -Wno-option-ignored)
+
+target_compile_options(compile_for_gpu
+    INTERFACE
+        -fno-gpu-rdc
+        -Wno-cuda-compat
+        -Wno-unused-command-line-argument
+        -Wno-invalid-command-line-argument
+        -Wno-option-ignored
+        -Xclang
+        -fallow-half-arguments-and-returns)
+
+target_link_libraries(compile_for_gpu INTERFACE hip::device)
 check_cxx_compiler_flag("--cuda-host-only -fhip-lambda-host-device -x hip" HAS_HIP_LAMBDA_HOST_DEVICE)
 
+if(WIN32)
+    target_compile_options(compile_for_gpu INTERFACE -fdelayed-template-parsing -fms-extensions -fms-compatibility)
+else()
+    target_compile_features(compile_for_gpu INTERFACE cxx_std_17)
+endif()
+
 if(HAS_HIP_LAMBDA_HOST_DEVICE)
     message(STATUS "Enable -fhip-lambda-host-device")
     target_compile_options(compile_for_gpu INTERFACE -fhip-lambda-host-device)
 endif()
 
 set_target_properties(migraphx_device PROPERTIES EXPORT_NAME device)
-rocm_set_soversion(migraphx_device ${MIGRAPHX_SO_VERSION})
-rocm_clang_tidy_check(migraphx_device)
-target_link_libraries(migraphx_device PUBLIC migraphx)
-target_link_libraries(migraphx_device PRIVATE compile_for_gpu)
+target_compile_options(migraphx_device
+    PRIVATE
+        -fno-gpu-rdc
+        -Xclang
+        -fallow-half-arguments-and-returns
+        -Wno-shadow-uncaptured-local
+        -Wno-cuda-compat
+        -Wno-unused-command-line-argument
+        -Wno-invalid-command-line-argument
+        -Wno-option-ignored)
+
+target_compile_features(migraphx_device PRIVATE cxx_std_17)
```

Comment:
Dont use `cxx_std_17`. We already set the C++ version at the top level.

Reply:
We must repeat it since it looks like that information is not propagated on Windows correctly by the CMake (or tools).

Reply:
The flags for c++17 should be added with `add_compile_options` at the top-level cmake, like we do here:

https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/CMakeLists.txt#L93

Also, we dont want to use `target_compile_features` since the version isnt stable.

Reply:
I see. The line you are referring to must have been removed while merging the branch `develop` to the branch `uif2/develop`. I cannot see it on the `uif2/develop` branch.